hal/common/retarget.cpp

```diff
@@ -138,14 +138,21 @@ static inline int openmode_to_posix(int openmode) {
 #endif
     return posix;
 }
+#if defined(__MICROLIB)
+extern "C" WEAK void mbed_sdk_init(void);
```

Comment:
isn't this declared already on line 527 - https://github.com/LMESTM/mbed/blob/fecc8e335ffeddaf5dbbfc28e657eaae123878b5/hal/common/retarget.cpp#L527